### PR TITLE
deb_src: introduce timeout when syncing DEB channels (bsc#1225960)

### PR DESCRIPTION
**Pull request by meaksh**
## What does this PR change?

As mentioned in the title, this PR introduces a timeout when running "reposync" against DEB channels. This would prevent getting reposync stuck in case connections with the remote repositories gets stuck.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: **no tests for this corner case**

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/24459

Port(s): https://github.com/SUSE/spacewalk/pull/24755

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: python/spacewalk/common/repo.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def get(self, key: typing.Any) -> typing.Optional[typing.Any]:  # type: ignore
                 )
             return self[key]
 
-    def __init__(self, url: str, proxies: dict = None, gpg_verify: bool = True):
+    def __init__(self, url: str, proxies: dict = None, gpg_verify: bool = True, timeout: int = None):
```
Comment (agraul) [inline python/spacewalk/common/repo.py:87]: `None` is not an `int`, either the type hint or the default value is wrong.

Comment (meaksh) [reply python/spacewalk/common/repo.py]: fixed

File: python/spacewalk/common/repo.py [hunk 2]
```diff
@@ -84,7 +84,13 @@ def get(self, key: typing.Any) -> typing.Optional[typing.Any]:  # type: ignore
                 )
             return self[key]
 
-    def __init__(self, url: str, proxies: dict = None, gpg_verify: bool = True):
+    def __init__(
+        self,
+        url: str,
+        proxies: dict = None,
+        gpg_verify: bool = True,
+        timeout: typing.Optional[int] = None,
```
Comment (agraul) [inline python/spacewalk/common/repo.py:92]: Is `int` the correct type? In the python-requests [documentation](https://docs.python-requests.org/en/latest/api/#requests.request), the `timeout` parameter is documented as follows:

> timeout ([float](https://docs.python.org/3/library/functions.html#float) or [tuple](https://docs.python.org/3/library/stdtypes.html#tuple)) – (optional) How many seconds to wait for the server to send data before giving up, as a float, or a [(connect timeout, read timeout)](https://docs.python-requests.org/en/latest/user/advanced/#timeouts) tuple.

Comment (meaksh) [reply python/spacewalk/common/repo.py:92]: In the context of reposync, the timeout is considered as integer value, so I guess that is what motivated the use of integer also for the timeout parameter of `DpkgRepo`  and `DebRepo` now.

There shouldn't be a problem passing this integer value to `requests` module, it should be evaluated properly.

**Review by agraul — CHANGES_REQUESTED** [review]
Please squash the second and third commit into the first. From my POV we can merge it afterwards.

**Comment by meaksh** [comment]
Thanks @agraul. I've squashed the commits as you suggested but I would need an explicit approval here as your review is still "requesting changes", so GitHub won't allow to merge this.

**Review by agraul — COMMENTED** [review]
I don't really think the new commit message containing the summaries of the squashed commits is useful. Are they included intentionally?

**Review by ycedres — APPROVED** [review]
LGTM.
